tools/rosgraph/src/rosgraph/roslogging.py

```diff
@@ -60,24 +60,33 @@ def findCaller(self, dummy=False): # Dummy second arg to match Python3 function
         if f is not None:
             f = f.f_back
         while hasattr(f, "f_code"):
-            # we search the right frame using the data already found by parent class
-            # following python logging findCaller() implementation logic
+            # Search for the right frame using the data already found by parent class.
             co = f.f_code
             filename = os.path.normcase(co.co_filename)
-            if filename != file_name or f.f_lineno != lineno or co.co_name != func_name:
-                f = f.f_back
-                continue
-            # we found  the correct frame, now extending func_name with class name
-            try:
-                class_name = f.f_locals['self'].__class__.__name__
-                func_name = '%s.%s' % (class_name, func_name)
-            except KeyError:  # if the function is unbound, there is no self.
-                pass
-            break
+            if filename == file_name and f.f_lineno == lineno and co.co_name == func_name:
+                break
+            f = f.f_back
+
+        # Jump up two more frames, as the logger methods have been double wrapped.
+        if f.f_back:
+            f = f.f_back
+        if f.f_back:
+            f = f.f_back
```

Comment:
That's a better idea. FWIW, I can't really see a case where it would be only one layer of wrapping, so the following _should_ be safe:

```
if f.f_code and f.f_code.co_name == '_base_logger':
    f = f.f_back.f_back
```